test/CMakeLists.txt

```diff
@@ -48,5 +38,33 @@ add_custom_target(installcheck
   TEST_INPUT_DIR=${TEST_INPUT_DIR}
   TEST_OUTPUT_DIR=${TEST_OUTPUT_DIR}
   PG_REGRESS=${PG_REGRESS}
-  ${CMAKE_CURRENT_SOURCE_DIR}/pg_regress.sh ${PG_REGRESS_OPTS}
+  )
+
+set(PG_REGRESS_OPTS_TEMP_INSTANCE
+  --user=${TEST_PGUSER}
+  --port=${TEST_PGPORT_TEMP_INSTANCE}
+  --temp-instance=${TEST_CLUSTER}
+  --temp-config=${TEST_OUTPUT_DIR}/postgresql.conf
+  ${PG_REGRESS_OPTS_COMMON}
+)
+
+set(PG_REGRESS_OPTS_LOCAL
+  --user=postgres
+  --port=${TEST_PGPORT_LOCAL}
+  ${PG_REGRESS_OPTS_COMMON}
+)
+
+# Configuration setting needed to LOAD module for tests instead of
+# preloading
+file(MAKE_DIRECTORY ${TEST_OUTPUT_DIR})
+file(WRITE ${TEST_OUTPUT_DIR}/postgresql.conf "shared_preload_libraries=timescaledb")
+
+add_custom_target(installcheck
+  COMMAND ${CMAKE_COMMAND} -E env ${PG_REGRESS_ENV}
+  ${CMAKE_CURRENT_SOURCE_DIR}/pg_regress.sh ${PG_REGRESS_OPTS_TEMP_INSTANCE}
+  USES_TERMINAL)
+
+add_custom_target(installchecklocal
```

Comment:
What's the difference here? This runs against an existing PostgreSQL instance? Could you add a comment here describing this target?

Reply:
fixed